query-engine/core/src/schema_builder/output_types/mutation_type.rs

```diff
@@ -54,24 +54,36 @@ fn create_nested_inputs(ctx: &mut BuilderContext) {
     while !(nested_create_inputs_queue.is_empty() && nested_update_inputs_queue.is_empty()) {
         // Create inputs.
         for (input_object, rf) in nested_create_inputs_queue.drain(..) {
-            let mut fields = vec![input_fields::nested_create_one_input_field(ctx, &rf)];
+            let mut fields = vec![];
 
-            append_opt(&mut fields, input_fields::nested_connect_input_field(ctx, &rf));
-            append_opt(&mut fields, input_fields::nested_connect_or_create_field(ctx, &rf));
+            if rf.related_model().supports_create_operation == true {
```

Comment:
I sense a pattern with your bools 😄 This one I do more firmly insist on keeping it as `if rf.related_model().supports_create_operation`.

Reply:
(same with any other follow ups)

Reply:
At least its not `===` ;-)

Reply:
`matches!(rf.related_model().supports_create_operation, true)`